src/components/menu/MenuOpener/MenuOpenedByHover.tsx

```diff
@@ -0,0 +1,54 @@
+// (C) 2007-2018 GoodData Corporation
+import * as React from 'react';
+import { IMenuOpenedBySharedProps } from './MenuOpenedBySharedProps';
+import MenuPosition from '../Position/MenuPosition';
+
+export default class MenuOpenedByHover extends React.Component<IMenuOpenedBySharedProps, {}> {
+    private static openCloseDelayMs = 200;
+
+    private timer: number = null;
+
+    public componentWillUnmount() {
+        this.clearTimers();
+    }
+
+    public render() {
+        return (
+            <MenuPosition
+                toggler={(
+                    <div onMouseEnter={this.hoverStart} onMouseLeave={this.hoverEnd}>
+                        {this.props.toggler}
+                    </div>
+                )}
+                opened={this.props.opened}
+                topLevel={this.props.topLevel}
+                direction={this.props.direction}
+                spacing={this.props.spacing}
+                offset={this.props.offset}
+                portalTarget={this.props.portalTarget}
+            >
+                <div onMouseEnter={this.hoverStart} onMouseLeave={this.hoverEnd}>
+                    {this.props.children}
+                </div>
+            </MenuPosition>
+        );
+    }
+
+    private clearTimers = () => {
```

Comment:
please use more specific name perhaps with the singular form of "timer"

Reply:
Renamed to `clearCloseDelayTimer`